processing/src/main/java/io/druid/segment/incremental/OffheapIncrementalIndex.java

```diff
@@ -241,10 +245,11 @@ protected Integer addToFacts(
         }
 
         final Integer rowIndex = indexIncrement.getAndIncrement();
+        
+        indexAndOffsets.add(new int[]{bufferIndex, bufferOffset});
```

Comment:
can you add a comment in the code explaining why this needs to be before adding to the facts?